Module1/todo_app/data/trello_items.py

```diff
@@ -65,14 +65,28 @@ def get_cards(self):
         list_cards = []
 
         for item in the_cards:
+            print(item)
+            print(f"Keys in item: {item.keys()}")
             print(f"{item['due']} and {type(item['due'])}")
             if item['due'] is not None:
                 print(f"ok {item['due']}")
-                due_temp = datetime.fromisoformat(item['due'])
-                due_date = due_temp.strftime("%Y-%m-%d %H:%M:%S")
+                if item['due']:
+                    ## due_temp = datetime.fromisoformat(item['due'])
+                    due_temp = item['due'][:-1]
+                    print(f"DUE_TEMP = {due_temp}")
+                    due_date = datetime.strptime(due_temp, "%Y-%m-%dT%H:%M:%S.%f")
+                # due_date = due_temp.strftime("%Y-%m-%d %H:%M:%S")
+                else:
+                    due_date = None
             else:
-                due_date = 'No Due Date Set'
-            temp_dict = {"name": item['name'], "DESC": item['desc'], "ID": item['id'], "due": due_date, "ListID": item['idList']}
+                due_date = None
```

Comment:
Again very minor, but a slight alternative here - because following the different if/else cases can quickly become difficult, could be:
```python
            due_date = None
            if item['due'] is not None and item['due']:
                print(f"ok {item['due']}")
                ## due_temp = datetime.fromisoformat(item['due'])
                due_temp = item['due'][:-1]
                print(f"DUE_TEMP = {due_temp}")
                due_date = datetime.strptime(due_temp, "%Y-%m-%dT%H:%M:%S.%f")
                # due_date = due_temp.strftime("%Y-%m-%d %H:%M:%S")
```
Note that I've combined the two if statements too, but I'll admit I'm not sure what case would occur that we'd have `item['due']` be `not None` but then fail the `item['due']` case. Could it be an empty string for example? If we expect either a valid date or a None, then we could potentially get rid of one of those cases